scripts/machine-launch-script.sh

```diff
@@ -1,82 +1,311 @@
+    #    37 | ebl_syscall_abi (Ebl *ebl, int *sp, int *pc, int *callno, int *args)
+    #       |                                                           ~~~~~^~~~
+    # In file included from ./../libasm/libasm.h:35,
+    #                  from ./libeblP.h:33,
+    #                  from ebl_syscall_abi.c:33:
+    # ./libebl.h:254:46: note: previously declared as an array 'int[6]'
+    #   254 |                             int *callno, int args[6]);
+    #       |                                          ~~~~^~~~~~~
+    #
+    # pin to gcc=10 until we get that fixed.
+
+    CONDA_PACKAGE_SPECS+=( gcc=10 gxx=10 binutils )
+
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync psutil doit gitpython humanfriendly e2fsprogs ctags bison flex expat )
+    # cross-compile glibc 2.28+ deps
+    CONDA_PACKAGE_SPECS+=( make )
+    # build-libelf wants autoconf
+    CONDA_PACKAGE_SPECS+=( autoconf automake libtool )
+    # other misc deps
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect \
+        dtc \
+        verilator==4.034 \
+    )
+
+    # python packages
+    # While it is possible to install using pip after creating the
+    # conda environment, pip's dependency resolution can conflict with
+    # conda and create broken environments.  It's best to use the conda
+    # packages so that the environment is consistent
+    CONDA_PACKAGE_SPECS+=( \
+        boto3==1.20.21 \
+        colorama==0.4.3 \
+        argcomplete==1.12.3 \
+        python-graphviz==0.19 \
+        pyparsing==3.0.6 \
+        numpy==1.19.5 \
+        kiwisolver==1.3.1 \
+        matplotlib==3.3.4 \
+        pandas==1.1.5 \
+        awscli==1.22.21 \
+        pytest==6.2.5 \
+        pytest-dependency==0.5.1 \
+        pytest-mock==3.7.0 \
+        moto==3.1.0 \
+        pyyaml==5.4.1 \
+        mypy==0.931 \
+        types-pyyaml==6.0.4 \
+        boto3-stubs==1.21.6 \
+        botocore-stubs==1.24.7 \
+        mypy-boto3-s3==1.21.0 \
+    )
+
+    if [[ "$CONDA_ENV_NAME" == "base" ]]; then
```

Comment:
I've made changes so that the script no longer installs the firesim deps into the base environment but instead defaults to creating a 'firesim' environment.  I'll be pushing that along with come cleanup of the docs.  

I've also added a `--reinstall-conda` option that people can use to reinstall the base environment if it gets broken.